x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -265,6 +266,54 @@ export function registerIndexRoutes({
     })
   );
 
+  router.post(
+    {
+      path: '/internal/enterprise_search/indices/{indexName}/ml_inference/pipeline_processors',
+      validate: {
+        params: schema.object({
+          indexName: schema.string(),
+        }),
+        body: schema.object({
+          pipeline_name: schema.string(),
+          model_id: schema.string(),
+          source_field: schema.string(),
+          destination_field: schema.maybe(schema.nullable(schema.string())),
+        }),
+      },
+    },
+
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const { client } = (await context.core).elasticsearch;
+      const {
+        model_id: modelId,
+        pipeline_name: pipelineName,
+        source_field: sourceField,
+        destination_field: destinationField,
+      } = request.body;
+
+      const createResult = await createMlInferencePipeline(
+        pipelineName,
+        modelId,
+        sourceField,
+        destinationField || modelId,
+        client.asCurrentUser
+      );
+
+      if (createResult.exists) {
+        return response.conflict({
+          body: new Error(`There is already a pipeline with name '${pipelineName}'.`),
```

Comment:
suggestion: message to use localisation

Reply:
I'll add a tech debt ticket for localization in order to keep the scope of this PR small.

Reply:
Filed https://github.com/elastic/enterprise-search-team/issues/2754 for this.